### PR TITLE
changed eslint config

### PR DESCRIPTION
**Pull request by rdss-hmorsi**
Hey Guys, we drafted our first version of our new eslint config. If you have any suggestions, wishes or something feel free to add a changerequest here :)

### CODE REVIEW

File: config/eslint/.eslintrc.js [hunk 1]
```diff
@@ -1,105 +1,96 @@
...
-        "indent": [
-            "error",
+    /**
+     * 0 = turned off
+     * 1 = warning
+     * 2 = error
+     */
+    'rules': {
+        'new-cap': 1,
+        'no-caller': 0,
```
Comment (rdss-dheuskel) [inline config/eslint/.eslintrc.js:43]: suggesting at least 1 here - ecmascript 5 in strict mode disallows the use of caller anyways

**Comment by rdss-dheuskel** [comment]
also suggesting adding the following:

no-implicit-globals (disallow declarations in the global scope)
no-return-assign (disallow assignment operators in `return` statements)
no-unused-expressions (disallow unused expressions)
radix (enforce the consistent use of the radix argument when using `parseInt()`)

with level warning at least.

**Comment by rdss-mhoevelmanns** [comment]
> i suggest also adding the following:
> 
> no-implicit-globals (disallow declarations in the global scope)
> no-return-assign (disallow assignment operators in `return` statements)
> no-unused-expressions (disallow unused expressions)
> radix (enforce the consistent use of the radix argument when using `parseInt()`)
> 
> with level warning at least.



> i suggest also adding the following:
> 
> no-implicit-globals (disallow declarations in the global scope)
> no-return-assign (disallow assignment operators in `return` statements)
> no-unused-expressions (disallow unused expressions)
> radix (enforce the consistent use of the radix argument when using `parseInt()`)
> 
> with level warning at least.



> i suggest also adding the following:
> 
> no-implicit-globals (disallow declarations in the global scope)
> no-return-assign (disallow assignment operators in `return` statements)
> no-unused-expressions (disallow unused expressions)
> radix (enforce the consistent use of the radix argument when using `parseInt()`)
> 
> with level warning at least.

I agree with all the rules except "no-implicit-globals". 
We cannot define "no-implicit-globals", because at least in the ZooRoyal project, "globals" are used very often.

**Comment by rdss-jpupeter** [comment]
> > i suggest also adding the following:
> > no-implicit-globals (disallow declarations in the global scope)
> > no-return-assign (disallow assignment operators in `return` statements)
> > no-unused-expressions (disallow unused expressions)
> > radix (enforce the consistent use of the radix argument when using `parseInt()`)
> > with level warning at least.
> 
> > i suggest also adding the following:
> > no-implicit-globals (disallow declarations in the global scope)
> > no-return-assign (disallow assignment operators in `return` statements)
> > no-unused-expressions (disallow unused expressions)
> > radix (enforce the consistent use of the radix argument when using `parseInt()`)
> > with level warning at least.
> 
> > i suggest also adding the following:
> > no-implicit-globals (disallow declarations in the global scope)
> > no-return-assign (disallow assignment operators in `return` statements)
> > no-unused-expressions (disallow unused expressions)
> > radix (enforce the consistent use of the radix argument when using `parseInt()`)
> > with level warning at least.
> 
> I agree with all the rules except "no-implicit-globals".
> We cannot define "no-implicit-globals", because at least in the ZooRoyal project, "globals" are used very often.

but just because we actually use this, dosen´t mean that we should continue using this.In my opinion, this rule would not harm us.

**Comment by rdss-dheuskel** [comment]
> > > i suggest also adding the following:
> > > no-implicit-globals (disallow declarations in the global scope)
> > > no-return-assign (disallow assignment operators in `return` statements)
> > > no-unused-expressions (disallow unused expressions)
> > > radix (enforce the consistent use of the radix argument when using `parseInt()`)
> > > with level warning at least.
> > 
> > 
> > > i suggest also adding the following:
> > > no-implicit-globals (disallow declarations in the global scope)
> > > no-return-assign (disallow assignment operators in `return` statements)
> > > no-unused-expressions (disallow unused expressions)
> > > radix (enforce the consistent use of the radix argument when using `parseInt()`)
> > > with level warning at least.
> > 
> > 
> > > i suggest also adding the following:
> > > no-implicit-globals (disallow declarations in the global scope)
> > > no-return-assign (disallow assignment operators in `return` statements)
> > > no-unused-expressions (disallow unused expressions)
> > > radix (enforce the consistent use of the radix argument when using `parseInt()`)
> > > with level warning at least.
> > 
> > 
> > I agree with all the rules except "no-implicit-globals".
> > We cannot define "no-implicit-globals", because at least in the ZooRoyal project, "globals" are used very often.
> 
> but just because we actually use this, dosen´t mean that we should continue using this.In my opinion, this rule would not harm us.

its not about using globals, its about defining them implicitly. you can still use globals if you assign them on the global object, this rule will not apply.

from https://eslint.org/docs/rules/no-implicit-globals:

```
/*eslint no-implicit-globals: "error"*/

// explicitly set on window
window.foo = 1;
window.bar = function() {};

```
this would be fine whereas

```
/*eslint no-implicit-globals: "error"*/

var foo = 1;

function bar() {}
```

this wouldnt.

**Comment by rdss-sknott** [comment]
ACHTUNG! Branch wurde gerebased! :information_source:

**Review by rdss-mhoevelmanns — APPROVED** [review]

**Review by rdss-egarro — APPROVED** [review]

**Comment by rdss-sknott** [comment]
Was ist das denn für ein force push gewessen?!

**Comment by rdss-dheuskel** [comment]
> Was ist das denn für ein force push gewessen?!

habe auf master rebased statt auf 2.x - reparatur kommt gleich.

**Review by rdss-smetzger — APPROVED** [review]

**Review by rdss-jpupeter — APPROVED** [review]

**Pull request merged**